colab-notebooks-examples/search/04-multilingual.ipynb

```diff
@@ -0,0 +1,487 @@
+   "outputs": [],
+   "source": [
+    "actions = []\n",
+    "for article in articles:\n",
+    "    actions.append({\"index\": {\"_index\": \"articles\"}})\n",
+    "    passage = article[\"passage\"]\n",
+    "    passageEmbedding = model.encode(f\"passage: {passage}\").tolist()\n",
+    "    article[\"passage_embedding\"] = passageEmbedding\n",
+    "    actions.append(article)\n",
+    "\n",
+    "client.bulk(index=\"articles\", operations=actions)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "MrBCHdH1u8Wd"
+   },
+   "source": [
+    "# Multilingual Semantic Search\n",
+    "\n",
+    "In the following, we will search using two kinds of queries:\n",
+    " * Query in English to find documents in any language\n",
+    " * Query in German to find documents in German only (using a filter),\n",
+    "   to show the model's capabilities in non-English languages\n",
+    "\n",
+    "Note again that the query is prefixed with \"query: \", which the model requires to encode the query properly.\n",
+    "\n",
+    "A quick translation for those unfamiliar with German:\n",
+    " * \"health\" -> \"Gesundheit\"\n",
+    " * \"wall\" -> \"Mauer\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pretty_response(response):\n",
+    "    for hit in response[\"hits\"][\"hits\"]:\n",
+    "        score = hit[\"_score\"]\n",
+    "        language = hit[\"_source\"][\"language\"]\n",
+    "        id = hit[\"_source\"][\"id\"]\n",
+    "        title = hit[\"_source\"][\"title\"]\n",
+    "        passage = hit[\"_source\"][\"passage\"]\n",
+    "        print()\n",
+    "        print(f\"ID: {id}\")\n",
+    "        print(f\"Language: {language}\")\n",
+    "        print(f\"Title: {title}\")\n",
+    "        print(f\"Passage: {textwrap.fill(passage, 120)}\")\n",
+    "        print(f\"Score: {score}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def query(q, language=None):\n",
+    "    knn = {\n",
+    "        \"field\": \"passage_embedding\",\n",
+    "        \"query_vector\" : model.encode(f\"query: {q}\").tolist(),\n",
+    "        \"k\": 2,\n",
+    "        \"num_candidates\": 5\n",
+    "    }\n",
+    "\n",
+    "    if language:\n",
+    "        knn[\"filter\"] = {\n",
+    "            \"term\": {\n",
+    "                \"language\": language,\n",
+    "            }\n",
+    "        }\n",
+    "\n",
+    "    return client.search(index=\"articles\", knn=knn)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pretty_response(query(\"health\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that in the results above, we see that the document about healthcare,\n",
```

Comment:
Might need another German speaker to verify the results 😄

Reply:
I am a German speaker 😉